src/atomate2/cp2k/jobs/core.py

```diff
@@ -0,0 +1,363 @@
+    stop_children_kwargs : dict
+        Keyword arguments that will get passed to :obj:`.should_stop_children`.
+    write_additional_data : dict
+        Additional data to write to the current directory. Given as a dict of
+        {filename: data}. Note that if using FireWorks, dictionary keys cannot contain
+        the "." character which is typically used to denote file extensions. To avoid
+        this, use the ":" character, which will automatically be converted to ".". E.g.
+        ``{"my_file:txt": "contents of the file"}``.
+    """
+
+    name: str = "static"
+    input_set_generator: Cp2kInputGenerator = field(default_factory=StaticSetGenerator)
+
+@dataclass
+class RelaxMaker(BaseCp2kMaker):
+    """
+    Maker to create CP2K relaxation jobs.
+
+    Parameters
+    ----------
+    name : str
+        The job name.
+    input_set_generator : .Cp2kInputGenerator
+        A generator used to make the input set.
+    write_input_set_kwargs : dict
+        Keyword arguments that will get passed to :obj:`.write_cp2k_input_set`.
+    copy_cp2k_kwargs : dict
+        Keyword arguments that will get passed to :obj:`.copy_cp2k_outputs`.
+    run_cp2k_kwargs : dict
+        Keyword arguments that will get passed to :obj:`.run_cp2k`.
+    task_document_kwargs : dict
+        Keyword arguments that will get passed to :obj:`.TaskDocument.from_directory`.
+    stop_children_kwargs : dict
+        Keyword arguments that will get passed to :obj:`.should_stop_children`.
+    write_additional_data : dict
+        Additional data to write to the current directory. Given as a dict of
+        {filename: data}. Note that if using FireWorks, dictionary keys cannot contain
+        the "." character which is typically used to denote file extensions. To avoid
+        this, use the ":" character, which will automatically be converted to ".". E.g.
+        ``{"my_file:txt": "contents of the file"}``.
+    """
+
+    name: str = "relax"
+    input_set_generator: Cp2kInputGenerator = field(default_factory=RelaxSetGenerator)
+
+@dataclass
+class CellOptMaker(BaseCp2kMaker):
+
+    name: str = "relax"
+    input_set_generator: Cp2kInputGenerator = field(default_factory=CellOptSetGenerator)
+
+@dataclass
+class HybridStaticMaker(BaseCp2kMaker):
+    """
+    Maker for static hybrid jobs
+
+    Parameters
+    ----------
+    name : str
+        The job name.
+    hybrid_functional : str
+        Built-in hybrid functional to use.
+    input_set_generator : .Cp2kInputGenerator
+        A generator used to make the input set.
+    """
+
+    name: str = "hybrid static"
+    hybrid_functional: str = "PBE0"
+    input_set_generator: Cp2kInputGenerator = field(default_factory=HybridStaticSetGenerator)
+
+    def __post_init__(self):
+        """Update the input settings with hybrid_functional attribute"""
+        self.input_set_generator.user_input_settings.update({"activate_hybrid": {"hybrid_functional": self.hybrid_functional}})
```

Comment:
So, let's say I go with option 1 and just trust the input set generator. Does that mean for the flows, like the one shown below, you would like to also remove `hybrid_functional` as an attribute? If we remove that, then the maker would have to be set by the user, and then the user would also have to set the input set generator inside of it. Just seems like a couple layers for something that is incredibly common, changing the hybrid functional. Or, do you think that the job should rely on the input set generator, but the flow can use an attribute to override things?
```
@dataclass
class HybridFlowMaker(Maker):
    hybrid_functional: str = "PBE0"
    initialize_with_pbe: bool = field(default=True)
    pbe_maker: Maker = field(default_factory=StaticMaker)
    hybrid_maker: Maker = field(default_factory=HybridStaticMaker)
```

Reply:
I think my main issue with this is overriding `user_input_settings` because it means the user can never override the job when I think we should strictly defer to the user if they are overriding.

Another alternative would be to override the input set config dict instead. That would achieve the same goal and still allow the user to override.

Reply:
Okay how about we delete the post-init for the jobs, but for flows, leave in the following:

```
def __post_init__(self):
    updates = {"activate_hybrid": {'hybrid_functional': self.hybrid_functional}}
    if self.initialize_with_pbe:
        updates['activate_hybrid'].update({"screen_on_initial_p": True, "screen_p_forces": True})
    self.hybrid_maker = update_user_input_settings(self.hybrid_maker, updates)
```

This does still do an override though.

Reply:
I made a further modification, which might help with the problem. `update_user_input_settings` is not modified to deal with nested dicts. This lets the powerup be called multiple times. The flow uses the powerup to set the necessary parameters, but users can still override with that powerup.

Reply:
Hey Nick, somehow I missed your comment.

From what I can see this current implementation will still suffer from the same issue I mentioned above, namely that this will be broken:

```
input_set = HybridStaticSetGenerator(
    user_input_settings={"activate_hybrid": {"hybrid_functional": "HSE"}}
)
maker = HybridStaticMaker(input_set_generator=input_set)
```

I think the best plan is to just update the config dict instead. That should fix all behaviour.

Reply:
If we update the config_dict, then `HybridStaticSetGenerator.get_input_updates()` will override it. So Job sets to HSE06, but then the set will override to the default PBE0...

Reply:
Alright. Now this should meet your specifications:

attributes have been dropped from the jobs, so to specify a job with HSE06 instead of default PBE0, you use the user_input_settings like you said

```python
input_set = HybridStaticSetGenerator(
    user_input_settings={"activate_hybrid": {"hybrid_functional": "HSE06"}}
)
maker = HybridStaticMaker(input_set_generator=input_set)
```

The flow now works with

```python
def __post_init__(self):
        """Initializing with PBE allows CP2K to screen exchange integrals using
        the PBE density matrix, which creates huge speed-ups. Rarely causes
        problems so it is done as a default here.
        """

        updates = {"activate_hybrid": {"hybrid_functional": self.hybrid_functional}}
        if self.initialize_with_pbe:
            updates["activate_hybrid"].update(
                {"screen_on_initial_p": True, "screen_p_forces": True}
            )
        self.hybrid_maker.input_set_generator.user_input_settings = recursive_update(
            updates, self.hybrid_maker.input_set_generator.user_input_settings
        )
```

Where config dict is left alone, so that get_input_updates doesn't override it, and instead we just call an update on the user input settings. As written `recursive_update` treats the settings from `__post_init__` as the base dict and treats the `user_input_settings` as the update dictionary. So whatever user specifies will be applied, but if nothing is supplied then it uses these flow settings.